waku/v2/protocol/filter/waku_filter.go

```diff
@@ -149,6 +149,13 @@ func (wf *WakuFilter) onRequest(s network.Stream) {
 func (wf *WakuFilter) pushMessage(subscriber Subscriber, msg *pb.WakuMessage) error {
 	pushRPC := &pb.FilterRPC{RequestId: subscriber.requestId, Push: &pb.MessagePush{Messages: []*pb.WakuMessage{msg}}}
 
+	// We connect first so dns4 addresses are resolved (NewStream does not do it)
+	err := wf.h.Connect(wf.ctx, wf.h.Peerstore().PeerInfo(subscriber.peer))
+	if err != nil {
+		wf.subscribers.FlagAsFailure(subscriber.peer)
```

Comment:
I added this line, since it was causing test failures without it. The test was ensuring that after a host was closed it would be marked as a failure, so this change feels like an oversight.

Reply:
Great minds think alike. Richard Ramos added the same fix yesterday. https://github.com/status-im/go-waku/pull/209/files